homeassistant/components/sensibo/climate.py

```diff
@@ -276,61 +276,44 @@ async def async_set_temperature(self, **kwargs) -> None:
         result = await self._async_set_ac_state_property(
             "targetTemperature", int(temperature)
         )
-        if result:
-            self.coordinator.data[self.unique_id]["target_temp"] = int(temperature)
-            self.async_write_ha_state()
+        await self.async_write_result(result, "target_temp", int(temperature))
 
     async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set new target fan mode."""
         result = await self._async_set_ac_state_property("fanLevel", fan_mode)
-        if result:
-            self.coordinator.data[self.unique_id]["fan_mode"] = fan_mode
-            self.async_write_ha_state()
+        await self.async_write_result(result, "fan_mode", fan_mode)
 
     async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         """Set new target operation mode."""
         if hvac_mode == HVAC_MODE_OFF:
             result = await self._async_set_ac_state_property("on", False)
-            if result:
-                self.coordinator.data[self.unique_id]["on"] = False
-                self.async_write_ha_state()
+            await self.async_write_result(result, "on", False)
             return
 
         # Turn on if not currently on.
         if not self.coordinator.data[self.unique_id]["on"]:
             result = await self._async_set_ac_state_property("on", True)
-            if result:
-                self.coordinator.data[self.unique_id]["on"] = True
+            await self.async_write_result(result, "on", True)
```

Comment:
There is a slight change of logic here: the previous code did not write to ha_state.
Can you confirm that the change is intended?

If it is unintended you can rewrite like this:
```python
            if result := await self._async_set_ac_state_property("on", True):
                self.coordinator.data[self.unique_id]["on"] = True
```

Reply:
I was thinking it doesn't matter writing state twice but I am reconsidering now as this could have some effects. Will rewrite